### DISCUSSION TITLE
PUBLIC meaning?

### DISCUSSION THREAD

**Discussion by Anonymous941** [NationalSecurityAgency/ghidra#6562] [General]
Each release's name is structured as `ghidra_*_PUBLIC_*.zip`.  Out of curiosity, what does `PUBLIC` refer to?  Is it because there are some in development features that aren't public, or because there are confidential parts of the program?

**Comment by astrelsky** [comment]
> Each release's name is structured as `ghidra_*_PUBLIC_*.zip`. Out of curiosity, what does `PUBLIC` refer to? Is it because there are some in development features that aren't public, or because there are confidential parts of the program?

When you build Ghidra from source you'll get something like `ghidra_11.2_DEV`.

As far as I can tell it's just a `PUBLIC` release build vs `DEV` build.

**Comment by ghidra1** [comment]
PUBLIC refers the the final release distributions placed on GitHub.  Development builds use DEV as you mentioned.  The application.config file gets modified during our build process which produces the final PUBLIC distribution.

**Reply by Anonymous941** [reply]
Got it, thanks for clarifying!

**Comment by ghidra1** [comment]
The version number does not mean much with DEV since 1000 different builds may have been produced with the same version number with different code modifications.  DEV implies an unknown state within a development cycle.
